core/modules/comment/comment.install

```diff
@@ -470,6 +470,35 @@ function comment_update_1007() {
   }
 }
 
+/**
+ * New setting for comment titles. Replacing comment_subject_field
+ * with comment_title_options.
+ */
+function comment_update_1008() {
+  foreach (config_get_names_with_prefix('node.type.') as $config_name) {
+    $config = config($config_name);
+    $comment_title = $config->get('settings.comment_subject_field');
+    if ($comment_title == COMMENT_TITLE_AUTO) {
+      // Set for auto generated titles.
+      $config->set('settings.comment_title_options', COMMENT_TITLE_AUTO);
+    } 
+    else {
+      // Set for custom titles.
+      $config->set('settings.comment_title_options', COMMENT_TITLE_CUSTOM);
+    }
+    $config->clear('settings.comment_subject_field');
+    $config->save();
+  }
+  
+  // Remove comment_subject_title variable now that it has been replaced.
+  foreach (node_type_get_types() as $type_name => $node_type) {
+    if (isset($node_type->settings['comment_subject_field'])) {
+      unset($node_type->settings['comment_subject_field']);
+    }
+    node_type_save($node_type);
```

Comment:
We should only use `config` calls in update hooks, as using APIs like `node_type_save()` will trigger hooks and may call more unexpected code that does not work correctly in the context of update.php